app/model/block.py

```diff
@@ -3,7 +3,20 @@ def __init__(self):
         self.id = None
         self.title = None
         self.sections = []
+        self.container = None
 
     def add_section(self, section):
         if section not in self.sections:
             self.sections.append(section)
+            section.container = self
+
+    def get_item_by_id(self, id):
```

Comment:
Consider renaming this to 'find' rather than 'get' to emphasise the fact that this is how it is operating.